File: dotnet/DesignPatterns/CreationalDesignPatterns/BuilderDesignPattern/BuilderDesignPattern.Example1/IReportBuilder.cs
namespace BuilderDesignPattern.Example1;

public interface IReportBuilder
{
    IReportBuilder SetTitle(string title);
    IReportBuilder SetContent(string content);
    IReportBuilder SetHeader(string header);
    IReportBuilder SetFooter(string footer);
    IReportBuilder SetFont(string font);
    Report Build();
}

File: nodejs/DesignPatterns/CreationalDesignPatterns/BuilderDesignPattern/BuilderDesignPattern.Example1/src/IReportBuilder.ts
import { Report } from './Report';

export interface IReportBuilder {
  setTitle(title: string): IReportBuilder;
  setContent(content: string): IReportBuilder;
  setHeader(header: string): IReportBuilder;
  setFooter(footer: string): IReportBuilder;
  setFont(font: string): IReportBuilder;
  build(): Report;
}
